src/transformers/models/kosmos2/processing_kosmos2.py

```diff
@@ -0,0 +1,696 @@
+                if with_bos:
+                    # for `<s>`
+                    mask = [0] + mask
+                # trailing part (which are not related to the image)
+                mask += [0] * (len(text_ids) - len(mask))
+                image_embeds_position_mask.append(mask)
+
+            if isinstance(text, list):
+                sorted_length = sorted([(idx, len(x)) for idx, x in enumerate(text_encoding.input_ids)])
+                _, min_len_not_padded = sorted_length[0]
+                idx, _ = sorted_length[-1]
+
+                text_encoding = self.tokenizer(
+                    text=[text[idx]],
+                    add_special_tokens=(add_special_tokens and add_eos_token),
+                    padding=padding,
+                    truncation=truncation,
+                    max_length=max_length,
+                    pad_to_multiple_of=pad_to_multiple_of,
+                    verbose=verbose,
+                    return_tensors=None,
+                    **kwargs,
+                )
+                max_len_padded = len(text_encoding.input_ids[0])
+
+                if min_len_not_padded != max_len_padded:
+                    if self.tokenizer.padding_side == "right":
+                        input_ids = [x + [self.tokenizer.pad_token_id] * (max_len_padded - len(x)) for x in input_ids]
+                        image_embeds_position_mask = [
+                            x + [0] * (max_len_padded - len(x)) for x in image_embeds_position_mask
+                        ]
+                        if "attention_mask" in encoding:
+                            encoding["attention_mask"] = [
+                                x + [0] * (max_len_padded - len(x)) for x in encoding["attention_mask"]
+                            ]
+                    elif self.tokenizer.padding_side == "left":
+                        input_ids = [[self.tokenizer.pad_token_id] * (max_len_padded - len(x)) + x for x in input_ids]
+                        image_embeds_position_mask = [
+                            [0] * (max_len_padded - len(x)) + x for x in image_embeds_position_mask
+                        ]
+                        if "attention_mask" in encoding:
+                            encoding["attention_mask"] = [
+                                [0] * (max_len_padded - len(x)) + x for x in encoding["attention_mask"]
+                            ]
+
+            # un-batch if necessary
+            if isinstance(text, str) and return_tensors is None:
+                input_ids = input_ids[0]
+                encoding["attention_mask"] = encoding["attention_mask"][0]
+                image_embeds_position_mask = image_embeds_position_mask[0]
+
+            # to the target tensor type
+            if return_tensors == "pt":
+                if not is_torch_available():
+                    raise RuntimeError("return_tensors set to 'pt' but PyTorch can't be imported")
+                import torch
```

Comment:
Put this on top implies that it will import torch (whenever it is available) even if an user don't want to use torch tensors as processor's outputs (say their want numpy or list). And if later we have TF/Flax port (sure we won't have it any soon), it will import all unnecessary TF/Flax.

I will keep as it is to align with our current design of `BatchFeature`. 

If we want to change this general design (I doubt it) in the future, we can easy change this file.

Let's not  ask a new design specific to a new model while all other places are still using the current design 🙏

Reply:
There is not a single processor in the entire library that has import torch in the processing because it does not re-implement the batch feature.